### ISSUE TITLE
Produce Fails: Producer.Produce() callback returns Local: Broker Unknown Topic error

### ISSUE THREAD

**Issue by harshpatel94** [edenhill/librdkafka#2888]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
I have been observing this issue quite while now. I have basic Kafka Producer which puts Key: Null and Value: {"xxxx:1234", "orderID"} as message body. 

Whenever I producer try to publish the message I get following error from Callback handler. 
`Broker: Unknown topic or partition`  

I see Topic is there. We use shared Enterprise Kafka Brokers and it is being able to connect to them easily. I tried to search everywhere, no luck. 

How to reproduce
================
- It is an intermittent issue. 80-90% works but rest 10-20% it does not. I would love to focus on those 10-20%, that why it happens even though topic is present. 

Checklist
=========
librdkafka - 1.4.0
Confluent.Kafka - 1.4.0
ProducerConfig: `{BootstrapServers="xxxxxx", CompressionType = ZSTD, clientID="XYZ"`
.NET Framework Version = v4.7.2
Running on server = 2008 R2 Windows Server

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<REPLACE with e.g., v0.10.5 or a git sha. NOT "latest" or "current">`
 - [ ] Apache Kafka version: `<REPLACE with e.g., 0.10.2.3>`
 - [ ] librdkafka client configuration: `<REPLACE with e.g., message.timeout.ms=123, auto.reset.offset=earliest, ..>`
 - [ ] Operating system: `<REPLACE with e.g., Centos 5 (x64)>`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

Any idea, what could be the reason behind this?

**Comment by edenhill** [comment]
produce() will return ERR__UNKNOWN_TOPIC if the topic metadata retrieved from the cluster indicates that the topic does not exist.
You can troubleshoot this further by setting `Debug: "topic,metadata"` to follow the topic's metadata state.

**Comment by chenhongluo** [comment]
@harshpatel94 Hi, I encountered the same problem as you. How do you solve it?

**Comment by harshpatel94** [comment]
@chenhongluo sorry to update this issue. If I remember correctly, we found out that it was an issue of configuration file on our side. Somehow our app was trying to search for an topic in different sets of brokers. Once we figured that out and modified the config, it started working. 

Try to make sure your topic is part of the brokerList which you are providing. Thank you @edenhill for better error messages to understand the issue. Thank you.
